**Bu alıştırmayı M.M.Coşkun'un Udemy platformu üzerindeki "Python : Sıfırdan İleri Seviye Programlama" kursunun pandas bölümünün ödevine cevap olarak yapıyorum.**

In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load in 

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the "../input/" directory.
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# Any results you write to the current directory are saved as output.

/kaggle/input/us-major-league-soccer-salaries/mls-salaries-2008.csv
/kaggle/input/us-major-league-soccer-salaries/mls-salaries-2009.csv
/kaggle/input/us-major-league-soccer-salaries/mls-salaries-2017.csv
/kaggle/input/us-major-league-soccer-salaries/mls-salaries-2007.csv
/kaggle/input/us-major-league-soccer-salaries/mls-salaries-2010.csv
/kaggle/input/us-major-league-soccer-salaries/mls-salaries-2015.csv
/kaggle/input/us-major-league-soccer-salaries/mls-salaries-2013.csv
/kaggle/input/us-major-league-soccer-salaries/mls-salaries-2011.csv
/kaggle/input/us-major-league-soccer-salaries/mls-salaries-2014.csv
/kaggle/input/us-major-league-soccer-salaries/mls-salaries-2016.csv
/kaggle/input/us-major-league-soccer-salaries/mls-salaries-2012.csv


In [2]:
data = pd.read_csv("../input/us-major-league-soccer-salaries/mls-salaries-2017.csv")#Verisetini okuma

In [3]:
data #Verisetine göz atma

club last_name    first_name position  base_salary  \
0    ATL   Almiron        Miguel        M    1912500.0   
1    ATL   Ambrose         Mikey        D      65625.0   
2    ATL      Asad         Yamil        M     150000.0   
3    ATL     Bloom          Mark        D      99225.0   
4    ATL  Carleton        Andrew        F      65000.0   
..   ...       ...           ...      ...          ...   
611  VAN  Tornaghi         Paolo       GK      80000.0   
612  VAN    Waston       Kendall        D     350000.0   
613  VAN  Williams       Sheanon        D     175000.0   
614  NaN   Babouli            Mo        F      54075.0   
615  NaN    Ramajo  David Mateos        D     420000.0   

     guaranteed_compensation  
0                 2297000.00  
1                   65625.00  
2                  150000.00  
3                  106573.89  
4                   77400.00  
..                       ...  
611                 80000.00  
612                368125.00  
613                184000.00  
614                 54075.00  
615                453333.33  

[616 rows x 6 columns]

In [4]:
data.head(10) # ilk 10 veri

club       last_name first_name position  base_salary  \
0  ATL         Almiron     Miguel        M    1912500.0   
1  ATL         Ambrose      Mikey        D      65625.0   
2  ATL            Asad      Yamil        M     150000.0   
3  ATL           Bloom       Mark        D      99225.0   
4  ATL        Carleton     Andrew        F      65000.0   
5  ATL         Carmona     Carlos        M     675000.0   
6  ATL           Garza       Greg        D     150000.0   
7  ATL  Gonzalez Pirez    Leandro        D     250008.0   
8  ATL          Goslin      Chris        M      70000.0   
9  ATL         Gressel     Julian        M      75000.0   

   guaranteed_compensation  
0               2297000.00  
1                 65625.00  
2                150000.00  
3                106573.89  
4                 77400.00  
5                725000.00  
6                150000.00  
7                285008.00  
8                 74000.00  
9                 93750.00

In [5]:
len(data.index) # Satır sayısı 

616

In [6]:
data["base_salary"].mean() # Ortalama Maaş

296977.73686688306

In [7]:
data["base_salary"].max() # En yüksek maaş

6660000.0

In [8]:
data[data["guaranteed_compensation"] == data["guaranteed_compensation"].max()] # En yüksek Tazminata sahip kişi

club last_name first_name position  base_salary  guaranteed_compensation
401  ORL      Kaka        NaN        M    6660000.0                7167500.0

In [9]:
data[data["guaranteed_compensation"] == data["guaranteed_compensation"].max()]["last_name"].iloc[0] ## En yüksek Tazminata sahip kişinin soyadi

'Kaka'

> **.iloc[]= Değerleri 0dan başlayarak indexler. Yukarıda tek bir değer olduğu için 401 numaralı index'in 0.index'i olan 'Kaka' adını verdi.** 

> > ****Gonzalez Pirez'in oynadığı mevki****

In [10]:
Gonz_position = soccer[data["last_name"] == "Gonzalez Pirez"]["position"].iloc[0]
if(Gonz_position == "F"):
    print("Forvet")
elif(Gonz_position == "M"):
    print("Orta Saha")
elif(Gonz_position == "D"):
    print("Defans")
else:
    print("Diğer")


NameError: name 'soccer' is not defined

In [11]:
data.groupby("position")["base_salary"].mean() #Pozisyona göre ortalama maaşlar

position
D      166574.093784
D-M    134930.000000
F      489588.598349
F-M    335834.000000
F/M    125000.000000
GK     146472.515538
M      376106.218632
M-D    219032.250000
M-F    212975.604211
M/F    550000.000000
Name: base_salary, dtype: float64

In [12]:
data["position"].nunique() #kaç farklı pozisyon(mevki) var.

10

In [13]:
data["position"].value_counts() # Her mevkideki oyuncu sayısı

M      212
D      185
F      109
GK      65
M-F     19
D-M     14
F-M      6
M-D      4
M/F      1
F/M      1
Name: position, dtype: int64

In [14]:
data["club"].value_counts()

VAN      32
ATL      31
PHI      31
CLB      30
ORL      30
DAL      29
SJ       29
NYRB     28
HOU      28
NYCFC    28
RSL      27
CHI      27
MTL      27
TOR      27
LA       27
POR      27
DC       27
MNUFC    27
COL      26
KC       26
SEA      25
NE       23
LAFC      2
Name: club, dtype: int64

In [15]:
#Soyadında 'SON' geçen futbolcuları bulur
def contains_word(last_name):
    return "son" in last_name
data[data["last_name"].apply(contains_word)]

club      last_name first_name position  base_salary  \
21     ATL       Peterson      Jacob        F    165300.00   
23     ATL       Robinson      Miles        D    125000.04   
45     CHI        Johnson     Daniel        M     65004.00   
48     CHI        Lampson       Matt       GK     76000.00   
85     CLB        Swanson        Ben        M     85000.00   
163     DC       Robinson      Jalen        D     69996.00   
233     LA       Jamieson   Bradford        F     66150.00   
265  MNUFC          Ibson        NaN        M    200004.00   
276  MNUFC       Thiesson     Jerome        D    171000.00   
295    MTL  Jackson-Hamel    Anthony        F     66150.00   
327     NE         Watson  Je-Vaughn      D-M    150000.00   
338  NYCFC       Harrison       Jack        M    130000.00   
341  NYCFC        Johnson       Sean       GK    220008.00   
400    ORL        Johnson       Will        M    414000.00   
442    PHI        Simpson        Jay        F    465000.00   
459    POR        Gleeson       Jake       GK    110000.00   
522    SEA       Svensson     Gustav        M    170000.04   
550     SJ       Thompson      Tommy        M    150000.00   
594    VAN       Jacobson     Andrew        M    150000.00   

     guaranteed_compensation  
21                 165300.00  
23                 195000.04  
45                  65004.00  
48                  81375.00  
85                 105416.67  
163                 69996.00  
233                 66150.00  
265                210337.40  
276                210166.67  
295                 66150.00  
327                155666.67  
338                165500.00  
341                220008.00  
400                450000.00  
442                508333.33  
459                115166.67  
522                170000.04  
550                155000.00  
594                175000.00